apps/src/netsim/NetSimRouterNode.js

```diff
@@ -1103,11 +1103,20 @@ NetSimRouterNode.prototype.forwardMessageToRecipient_ = function (message, onCom
 
   // TODO: Handle bad state where more than one wire matches dest address?
 
+  // Normally the recipient simulates a message.
```

Comment:
"Who simulates" effectively means, "which client has permission to pull this message out of storage?"  In a P2P situation, that must be the receiving client as the sender has no indication of whether the message has been read.

We could reverse the logic when a router is present, and have a client act as the router for all messages destined for their own node; that would allow the router to just pass-through the simulatedBy property for all messages it processes, but I'm afraid it would make other router features like limited bandwidth and router stats seem less accurate - since the router gets immediate local notification when the sender simulates, it can at least update its own logs and timings immediately, and put the network delay between the router and the recipient (matching the delay in the arrival of the router stats).